core/src/main/java/org/opensearch/sql/planner/physical/HighlightOperator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.physical;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Collections;
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.expression.Expression;
+
+@EqualsAndHashCode(callSuper = false)
+@Getter
+public class HighlightOperator extends PhysicalPlan {
+  @NonNull
+  private final PhysicalPlan input;
```

Comment:
Why do you store a parent class instance there?
I think you can call `super.next()` and `super.hasNext()` instead of `input.next()` and `input.hasNext()` and so on.

Reply:
Seems to be standard considering other operators also store the parent class instance like this.